### PR TITLE
Add createUserDoctor mutation & minor type FIX

### PR DESCRIPTION
**Pull request by konpap99**
Add mutation to implement doctor registration

 Example

mutation CreateUserDoctor($input: UserDoctorInput!){
  createUserDoctor(input: $input) {
    id
  }
}

Variables:
{
  "input": {
		"email": "konpap99@hotmail.com",
    "firstName": "Konstantinos",
    "lastName": "Papaioannou",
    "password": "12345",
    "address": {
      "streetName": "Josef-Wirth-Weg",
      "streetNumber": 21,
      "city": "Munich",
      "zipCode": 80939
    },
    "phoneNumber": "+30 6971947123",
  	"webpage": "https://konpap99.com",
    "specialities": [
      "Cardiologist",
      "Pathologist"
    ],
    "offeredSlots": [
      {"day": "Monday", "slotStart": "17:00", "slotStop": "18:00"},
      {"day": "Tuesday", "slotStart": "17:30", "slotStop": "18:30"}
    ]
  }
}

image

-----------------------------------------------------------------------
Notes: 
A transaction is used in case one of the save() is not successful the database rolls back

If a shim is needed please tell me. In this case I thought it wasn't useful because the registration of a user happens only once in our app

I would prefer a better error handling approach in mutation.ts. I would prefer not to throw 'Error' but provide something more meaningful. In any case the code should never be successful in the if statement

### CODE REVIEW

File: src/typeDefs.ts [hunk 1]
```diff
@@ -120,8 +120,8 @@ export const typeDefs = gql`
 
     type OfferedSlot {
         day: Weekday!
-        start: Time!
-        end: Time!
```
Comment (nerdsupremacist) [inline src/typeDefs.ts:124]: nit: So we're no longer using time here? Do we still need the type?

**Review by nerdsupremacist — APPROVED** [review]
lgtm

File: src/typeDefs.ts (hunk 1 above)
Comment (konpap99) [reply src/typeDefs.ts:124]: I don't think we need it

Comment (konpap99) [reply src/typeDefs.ts:124]: i ll fix it and merge it

**Review by Pellumbengineer — APPROVED** [review]
looks good!

**Pull request merged**